static/app/components/charts/optionSelector.tsx

```diff
@@ -67,7 +68,11 @@ class OptionSelector extends Component<Props, State> {
             {({isOpen, getMenuProps, getActorProps}) => (
               <Fragment>
                 <StyledDropdownButton {...getActorProps()} size="zero" isOpen={isOpen}>
-                  {selectedOption.label}
+                  <Truncate
+                    value={String(selectedOption.label)}
+                    maxLength={60}
+                    expandable={false}
+                  />
```

Comment:
Nit: you might be able to use a combination of `max-width: ...` and `text-overflow: ellipsis` here to dynamically set the amount of visible text.